### DISCUSSION TITLE
mobile hosting

### DISCUSSION THREAD

**Discussion by Yandurva08** [jagrosh/MusicBot#1349] [Questions] [answered]
Is there any way to host on mobile for free?

**Comment by Lastorder-DC** [comment, marked as answer]
Before continue,
1. This works only android phone(or jailbroken ios device - search for how to run java)
2. You have to know how to download files at linux shell, and edit files from shell

=====
1. install [termux](https://f-droid.org/ko/packages/com.termux/)
2. install java by `pkg install openjdk-17`
3. Follow Guides from step 2 - https://jmusicbot.com/setup/#2-download-jmusicbot
4. Hint: use wget to download jar file. use nano to edit config files.

**Reply by Yandurva08** [reply]
my Android version is a bit outdated
(Android 6.0 32bits)

**Reply by Lastorder-DC** [reply]
https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

p.s.) looks like java is not exist on old version, you might have to search for how to install java on termux android 5 release.
